linkatos/parser.py

```diff
@@ -7,48 +7,69 @@ def is_empty(message_list):
     return ((message_list is None) or (len(message_list) == 0))
 
 
+def capture_reaction(sub_message):
+    parsed = {}
+    parsed['message'] = sub_message['reaction']
+    parsed['channel'] = sub_message['item']['channel']
+    parsed['item_ts'] = sub_message['item']['ts']
+    parsed['type'] = 'reaction'
+    parsed['user'] = sub_message['user']
+    parsed['item_user'] = sub_message['item_user']
+    return parsed
+
+
+def capture_url(sub_message, url):
+    parsed = {}
+    parsed['message'] = url
+    parsed['channel'] = sub_message['channel']
+    parsed['ts'] = sub_message['ts']
+    parsed['type'] = 'url'
+    parsed['user'] = sub_message['user']
+    return parsed
+
+
 def parse(input_message, BOT_ID):
     """
         The Slack Real Time Messaging API is an events firehose.
         this parsing function returns None unless:
         1. someone posts a url starting with httpS?//
            in this case it returns the url and a out_type = 'url'
-        2. a yes or a no after a url
-           in this case it returns True for a yes, False for a no
-        The function also extracts the name of the channel
+        2. someone adds a thumbsup or a thumbsdown
+           in this case it returns the reaction and the type of outcome
     """
-    print(input_message)  # print the list of outputs to get them on screen
+    print('input_message:', input_message)  # print the list of outputs to get them on screen
 
     # default outcome
-    parsed = {'out': None, 'channel': None, 'type': None}
+    parsed = {'message': None,
+              'channel': None,
+              'ts': None,
+              'item_ts': None,
+              'type': None,
+              'user': None,
+              'item_user': None}
```

Comment:
It avoids issues of undefined variables when the bot starts running and messages are empty.
Maybe not the best way to solve the issue?